src/js/renderables/enemies/enemy.js

```diff
@@ -94,76 +84,56 @@ class Enemy extends me.Entity {
         }
     }
 
-    syncHitBox() {
-        this.hitbox.pos.x = this.pos.x;
-        this.hitbox.pos.y = this.pos.y;
-    }
-
     // Method to reduce the enemy's health when it takes damage
     takeDamage(damage) {
         this.health -= damage;
-
-        // If the health drops to 0 or below, call the die method
         if (this.health <= 0) {
             this.die();
         }
     }
 
     // Method to handle enemy death
     die() {
-        if(this.health <= 0) {
-
-        // Remove the enemy from the game world when it dies
-            this.onDestroy()
-
+        if(this.alive) {
             this.rewardPlayer();
-
-            console.log(`${this.element} enemy has been defeated!`);
+            // console.log(`${this} enemy has been defeated!`);
+            this.onDestroy()
         }
     }
 
     // Method to reward player on enemy death
     rewardPlayer(){
         applicationState.data.currency += this.reward;
-        this.updateCurrency();
-
         console.log(`Player rewarded with ${this.reward} coins.`);
     }
 
+    // Remove the enemy from the world when it dies
+    onDestroy() {
+        if (this.alive) {
+            console.log(`${this} enemy is being removed from the game world.`);
+            me.game.world.removeChild(this); 
+        }
+        this.alive = false;
```

Comment:
Remove this.alive = false since if we destroy/remove it, we're referencing a null object.

Reply:
this.alive is a built in attribute of melonJS's entity and has a default value of true--[Link](https://melonjs.github.io/melonJS/classes/Entity.html#alive)

I am not sure why, but I believe the removeChild method doesn't actually destroy the object, it just makes it not visible to the user. I can't find a method to free the resource (ie something like a destroy() method). So if I remove it, it causes the problem where the game will try to delete it multiple times.